exporters/stats/stackdriver/src/main/java/io/opencensus/exporter/stats/stackdriver/StackdriverExporterWorkerThread.java

```diff
@@ -60,31 +56,44 @@
     setName("ExportWorkerThread");
   }
 
+  // Registers the given view to Stackdriver Monitoring if the view is not registered yet, otherwise
+  // does nothing. Throws IllegalArgumentException if the given view is a different view that has a
+  // same name with a registered view.
+  @VisibleForTesting
   void registerView(View view) {
-    synchronized (monitor) {
-      View existing = registeredViews.get(view.getName());
-      if (existing != null) {
-        if (existing.equals(view)) {
-          // Ignore views that are already registered.
-          return;
-        } else {
-          throw new IllegalArgumentException(
-              "A different view with the same name is already registered: " + existing);
-        }
+    View existing = registeredViews.get(view.getName());
+    if (existing != null) {
+      if (existing.equals(view)) {
+        // Ignore views that are already registered.
+        return;
+      } else {
+        // TODO(songya): Do we still need to check this? Or can we assume that
+        // ViewManager.getAllExportedViews() will always return correct results?
+        throw new IllegalArgumentException(
+            "A different view with the same name is already registered: " + existing);
```

Comment:
I think it is better to handle this case, if possible, since there could be a version mismatch between the exporter and the API, and the API could support unregistering a view.  Does Stack Driver allow a view to be changed or unregistered?  If not, I think it would at least be better to log an error and ignore the view.

Reply:
Stackdriver doesn't allow changing a `View`, if we upload a `MetricDescriptor` (equivalent to `View`) which has the same name but different attributes to Stackdriver, it will throw an exception. Stackdriver does support unregistering (delete) a `MetricDescriptor`, though.

Reply:
I've removed the TODO and left a comment explaining this.

Reply:
I still think it would be better to just log an error and skip the view, if you don't think we should unregister and re-register the MetricDescriptor.  Currently, a view mismatch would cause export thread to stop running.

Reply:
I see, I've updated to use log instead. This also matches the behavior of trace exporter: https://github.com/census-instrumentation/opencensus-java/blob/master/impl_core/src/main/java/io/opencensus/implcore/trace/export/SpanExporterImpl.java#L136.